dev/tools/update_versions.dart

```diff
@@ -0,0 +1,194 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Updates the version numbers of the Flutter repo.
+// Only tested on Linux.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+
+const String kIncrement = 'increment';
+const String kBrokeSdk = 'broke-sdk';
+const String kBrokeFramework = 'broke-framework';
+const String kBrokeTest = 'broke-test';
+const String kBrokeDriver = 'broke-driver';
+const String kMarkRelease = 'release';
+const String kHelp = 'help';
+
+const String kYamlVersionPrefix = 'version: ';
+const String kDev = '-dev';
+
+void main(List<String> args) {
+  // If we're run from the `tools` dir, set the cwd to the repo root.
+  if (path.basename(Directory.current.path) == 'tools')
+    Directory.current = Directory.current.parent.parent;
+
+  final ArgParser argParser = new ArgParser();
```

Comment:
https://www.dartdocs.org/documentation/args/0.13.7/args/ArgParser/allowTrailingOptions.html is the one true way to ArgParse. :p